tests/test_api.py

```diff
@@ -137,6 +138,22 @@ def test_login(self):
             cl.token = uuid4()
             assert cl.default_headers() != {}
 
+    def test_get_user(self):
+        """Test that when a client is created, it logs in just once."""
+        with mock.patch.object(config, '_CONFIG', self.config):
+            self.assertEqual(config.get_config(), self.config)
+            client = api.Client
+            client.login = MagicMock()
+            response = MagicMock(
+                json=MagicMock(
+                    return_value={
+                        'username': 'admin'}))
+            client.request = MagicMock(return_value=response)
+            cl = client()
+            u = cl.get_user().json()['username']
+            assert u == config.get_config()['qcs']['username']
+            assert client.request.call_count == 1
```

Comment:
The mock object has some assert methods like the `assert_called_once_with`, which can help in cases like this when you also wanna evaluate the parameters received by the call.

More info at: https://docs.python.org/3.5/library/unittest.mock.html#unittest.mock.Mock.assert_called_once_with

Reply:
Cool, thanks! I used this.